pkg/controller/container-runtime-config/helpers.go

```diff
@@ -46,24 +50,43 @@ type tomlConfigStorage struct {
 	} `toml:"storage"`
 }
 
-// tomlConfig is another way of looking at a Config, which is
+// tomlConfigCRIOLogLevel is used for conversions when log-level is changed
 // TOML-friendly (it has all of the explicit tables). It's just used for
 // conversions.
-type tomlConfigCRIO struct {
+type tomlConfigCRIOLogLevel struct {
 	Crio struct {
-		crioconfig.RootConfig
-		API     struct{ crioconfig.APIConfig }     `toml:"api"`
-		Runtime struct{ crioconfig.RuntimeConfig } `toml:"runtime"`
-		Image   struct{ crioconfig.ImageConfig }   `toml:"image"`
-		Network struct{ crioconfig.NetworkConfig } `toml:"network"`
-		Metrics struct{ crioconfig.MetricsConfig } `toml:"metrics"`
+		Runtime struct {
+			LogLevel string `toml:"log_level,omitempty"`
+		} `toml:"runtime"`
 	} `toml:"crio"`
 }
 
-// ignitionConfig is a struct that holds the filepath and date of the various configs
+// tomlConfigCRIOPidsLimit is used for conversions when pids-limit is changed
+// TOML-friendly (it has all of the explicit tables). It's just used for
+// conversions.
+type tomlConfigCRIOPidsLimit struct {
+	Crio struct {
+		Runtime struct {
+			PidsLimit int64 `toml:"pids_limit,omitemtpy"`
+		} `toml:"runtime"`
+	} `toml:"crio"`
+}
+
+// tomlConfigCRIOLogSizeMax is used for conversions when log-size-max is changed
+// TOML-friendly (it has all of the explicit tables). It's just used for
+// conversions.
+type tomlConfigCRIOLogSizeMax struct {
+	Crio struct {
+		Runtime struct {
+			LogSizeMax int64 `toml:"log_size_max,omitempty"`
+		} `toml:"runtime"`
+	} `toml:"crio"`
+}
+
+// fileConfig is a struct that holds the filepath and date of the various configs
 // Using a struct array ensures that the order of the ignition files always stay the same
 // ensuring that double MCs are not created due to a change in the order
-type ignitionConfig struct {
+type fileConfig struct {
```

Comment:
(Absolutely non-blocking, I’m sure everybody loves name bike-shedding…: to me this reads as “configuration of a file”. `configFile`? `generatedFile`?)

Reply:
Ohh those are better suggestions. :smile:

Reply:
renamed it to `generatedConfigFile` so we can all be happy :D